src/main/java/mate/academy/service/impl/CinemaHallServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.CinemaHall;
+import mate.academy.service.CinemaHallService;
+
+@Service
+public class CinemaHallServiceImpl implements CinemaHallService {
+    @Inject
+    private CinemaHallDao cinemaHallDao;
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        return cinemaHallDao.add(cinemaHall);
+    }
+
+    @Override
+    public Optional<CinemaHall> get(Long id) {
+        return Optional.of(cinemaHallDao.get(id).get());
```

Comment:
Directly calling `.get()` on an `Optional` without checking if a value is present can lead to `NoSuchElementException`. The service layer should return the entity directly, not wrapped in an `Optional`. Consider handling these cases more gracefully, perhaps by throwing a custom exception if the `Optional` is empty.